tools/Customization/DevHome.Customization/ViewModels/Environment/DevDriveOptimizerCardViewModel.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Services;
+using DevHome.Customization.Views;
+using Microsoft.UI.Xaml.Controls;
+
+namespace DevHome.Customization.ViewModels.Environments;
+
+/// <summary>
+/// View model for the card that represents a dev drive optimizer on the dev drive insights page.
+/// </summary>
+public partial class DevDriveOptimizerCardViewModel : ObservableObject
+{
+    public string CacheToBeMoved { get; set; }
+
+    public string DevDriveOptimizationSuggestion { get; set; }
+
+    public string ExistingCacheLocation { get; set; }
+
+    public string ExampleLocationOnDevDrive { get; set; }
+
+    public string EnvironmentVariableToBeSet { get; set; }
+
+    public string OptimizerDevDriveDescription { get; set; }
+
+    /// <summary>
+    /// User wants to optimize a dev drive.
+    /// </summary>
+    [RelayCommand]
+    private async Task OptimizeDevDriveAsync(object sender)
+    {
+        var settingsCard = sender as Button;
+        if (settingsCard != null)
+        {
+            var optimizeDevDriveDialog = new OptimizeDevDriveDialog(CacheToBeMoved, ExistingCacheLocation, EnvironmentVariableToBeSet);
+            optimizeDevDriveDialog.XamlRoot = settingsCard.XamlRoot;
+            optimizeDevDriveDialog.RequestedTheme = settingsCard.ActualTheme;

```

Comment:
I don't think you want to do this -- it will use the right theme without it, and if the RequestedTheme is Default, then this will cause a bug when the System theme is changed. For example, if RequestedTheme is Default and ActualTheme is Dark, but then you set the system theme to Light, the theme of the dialog will stay Dark

Reply:
Without this, I see light theme dialog pop up for a dark themed app.

Reply:
Does it make a difference if you include the Style on the dialog like I mentioned below? There should be a way we can remove this.

Reply:
I am using Style="{StaticResource DefaultContentDialogStyle}" in the dialog xaml. Is that what you are refering to?

Reply:
I have the following in the dialog .Style="{StaticResource DefaultContentDialogStyle}" in the dialog.

That is not making any difference, if that is what you are referring to.